lib/v2/trending/allTrending.js

```diff
@@ -0,0 +1,165 @@
+                }))
+            ),
+        }))
+    );
+    for (const i of data) {
+        i.count = i.data.reduce((acc, { data }) => acc + data.length, 0);
+    }
+
+    return data.filter(({ count }) => count > 0);
+};
+
+// Generate Feed Items
+const searchLinkUrls = (keyword) => [
+    `https://tophub.today/search?e=tophub&q=${keyword}`,
+    `https://www.baidu.com/s?wd=${keyword}`,
+    `https://www.google.com/search?q=${keyword}`,
+    `https://www.zhihu.com/search?type=content&q=${keyword}`,
+    `https://s.weibo.com/weibo/${keyword}`,
+    `https://www.douyin.com/search/${keyword}`,
+    `https://so.toutiao.com/search?keyword=${keyword}`,
+];
+
+const searchLinkNames = ['热榜', '百度', '谷歌', '知乎', '微博', '抖音', '头条'];
+
+const createItem = ({ dateTime, data, count }, keywords, isToday) => ({
+    title: `${keywords.join(', ')} | ${dateTime.format(DATE_FORMAT)} 热点追踪 (${count})`,
+    author: 'Trending All In One',
+    pubDate: isToday ? new Date() : dateTime.endOf('day').toDate(),
+    description: art(path.join(__dirname, 'templates/content.art'), {
+        data,
+        queries: keywords.map((query) => ({
+            links: searchLinkUrls(encodeURIComponent(query)).map((url, index) => `<a href="${url}" rel="noopener noreferrer" target="_blank">${searchLinkNames[index]}</a>`),
+            key: query,
+        })),
+    }),
+    guid: `trending-all-in-one-${dateTime.toISOString()}-${keywords.join('-')}`,
+});
+
+// Main
+module.exports = async (ctx) => {
+    // Prevent making over 100 requests per invocation
+    if (ctx.params.numberOfDays > 14) {
+        throw new Error('days must be less than 14');
+    }
+    const numberOfDays = ctx.params.numberOfDays || 3;
+    const currentShanghaiDateTime = dayjs(toShanghaiTimezone(new Date()));
+    const currentShanghaiDateStr = currentShanghaiDateTime.format(DATE_FORMAT);
+    const dateList = [];
+    for (let i = 0; i < numberOfDays; i++) {
+        const d = currentShanghaiDateTime.subtract(i, 'day');
+        dateList.push(d);
+    }
+
+    const keywordList = ctx.params.keywords
+        .replace('，', ',')
+        .split(',')
+        .map((keyword) => keyword.trim());
+    const keywordStr = keywordList.join(', ');
+
+    const data = await fetchAllData(keywordList, dateList, ctx.cache);
+    const item =
+        data.length > 0
+            ? data.map((i, index) => createItem(i, keywordList, index === 0))
+            : [
+                  {
+                      title: `${keywordStr} | ${currentShanghaiDateStr} 热点追踪 (0)`,
+                      author: 'Trending All In One',
+                      description: `近${numberOfDays}日的热搜都不包含下列关键词：${keywordStr}<br>请耐心等待，或添加更多关键词试试。`,
+                      guid: `trending-all-in-one-${currentShanghaiDateTime.toISOString()}-${keywordStr}`,
+                      pubDate: new Date(),
```

Comment:
```suggestion
```
Do not add pubDate if there isn't a date.
docs: https://docs.rsshub.app/joinus/pub-date.html#mei-you-ri-qi